src/pybind/mgr/prometheus/module.py

```diff
@@ -551,35 +525,40 @@ def collect(self):
                 # Represent the long running avgs as sum/count pairs
                 if counter_info['type'] & self.PERFCOUNTER_LONGRUNAVG:
                     _path = path + '_sum'
-                    self.metrics.add_metric(_path, Metric(
-                        stattype,
-                        _path,
-                        counter_info['description'] + ' Total',
-                        ("ceph_daemon",),
-                    ))
-                    self.metrics.append(_path, value, (daemon,))
+                    if _path not in self.metrics:
```

Comment:
Minor nit: The ```Total``` and ```Count``` metrics could be setup in the same if block. I assume they always come in this pair? That would save one conditional in line 538.

Reply:
It would not really save the conditional, we would have to use `_path1 not in self.metrics or _path2 not in self.metrics` to be sure that the conditional is working properly (*). Fortunately, both of these are simple hash table lookups so it is not too bad performance-wise.

(*) I just have a bad feeling that someone somewhere will change a `<stg>_count` or `<stg>_sum` perf counter into a histogram that is represented by the long running avgs as a pair of `<stg>_sum` and `<stg>_count` and the code would break. It would also be pretty difficult to debug if it really happened.